pkg/volume/local/local.go

```diff
@@ -154,6 +187,25 @@ func (plugin *localVolumePlugin) ConstructVolumeSpec(volumeName, mountPath strin
 	return volume.NewSpecFromPersistentVolume(localVolume, false), nil
 }
 
+func (plugin *localVolumePlugin) ConstructBlockVolumeSpec(podUID types.UID, volumeName,
+	mapPath string) (*volume.Spec, error) {
+	block := v1.PersistentVolumeBlock
+	localVolume := &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: volumeName,
+		},
+		Spec: v1.PersistentVolumeSpec{
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				Local: &v1.LocalVolumeSource{
+					Path: "",
```

Comment:
Seems both ConstructBlockVolumeSpec() and ConstructVolumeSpec() can't construct proper volume spec. Any reason for this?

Reply:
Fixed ConstructBlockVolumeSpec() to specify the device path.  Regarding ConstructVolumeSpec(), @msau42 mentioned that it was not done because the device path was not needed.

Reply:
@dhirajh 

>>ConstructVolumeSpec(), @msau42 mentioned that it was not done because the device path was not needed.

Could you let me know the detail? I couldn't find her comment.

Reply:
It was from a chat, that I had a week back.   @msau42 please do correct me if I missed something. The path is not needed for unmounting.  The only thing the local volume plugin does is unmount the kubelet's pod volume bind mount. It leaves the real path unchanged.

Reply:
@mtanino local volumes are very simple and don't need the underlying path to do teardown.

For filesystem volumes: Setup will bind mount the path to the kubelet's volume directory.  Teardown only needs the kubelet's volume directory to unmount it.

For block volumes: I think it should be similar.  Setup returns the path.  Kubelet will create the appropriate mappings for it.  For teardown, the plugin doesn't need to do anything, just relies on kubelet to unmap everything.  Maybe one problem here is if the reconstruction also calls setup?  Then we do not have the underlying path to return.  We should probably fail or return an empty path.  Does the new reconstruction design that only does teardown help with this?

Reply:
@msau42 

>> local volumes are very simple and don't need the underlying path to do teardown.

I see. So during reconstruction, local volume plugin doesn't need to return the `Path` into reconstruct volume spec for both `filesystem` and `block`, right?

>>Maybe one problem here is if the reconstruction also calls setup? Then we do not have the >>underlying path to return. We should probably fail or return an empty path. Does the new
>>reconstruction design that only does teardown help with this?

IIUC, volume reconstruction doesn't call setUp(). It calls following methods during reconstruction.
- plugin's constructVolumeSpec()
- kubelet's GenerateUnmountVolumeFunc()
- kubelet's GenerateUnmountDeviceFunc()
   => plugin's TearDown()

(*) block volume case is similar to above steps.

In the new reconstruction design, @jingxu97 is proposing two clean up methods based on reconstruction method support by plugin.
1. reconciler updates volume information using reconstruction into the states and reconciler takes care of volume cleanup
2. reconciler cleans up the mounts directly using tearDown() without reconstruction

Either way, setUp() isn't called. Which way would prefer for `local volume plugin`?

Reply:
Yeah so if only TearDown is called, then local volume is fine with a blank spec.  Only volume name needs to be correctly populated.  I would prefer having the cleanup down by reconciler so that it can retry if it fails the first time.

Reply:
yes, agree. Added comment on reconstruction design doc.